README.md

```diff
@@ -1,289 +1,590 @@
-
-Since the reference implementation for most of this specification is the [Service Binding Operator](https://github.com/redhat-developer/service-binding-operator) we will be using the `ServiceBinding` CRD, which resides in [this folder](https://github.com/redhat-developer/service-binding-operator/tree/master/deploy/crds), as the entity that holds the binding request.  
-
-**Note** - a few updates / enhancements are being proposed to the current `ServiceBinding` CR, tracked [here](https://github.com/application-stacks/service-binding-specification/issues/16#issuecomment-605309629).
-
-Sample CR
+## Resource Type Schema
+
+```yaml
+status:
+  binding:
+    name:  # string
+```
+
+## Example Resource
+
+```yaml
+...
+status:
+  ...
+  binding:
+    name: production-db-secret
 ```
+
+## Well-known Secret Entries
+
+Other than the required `kind` entry and the recommended `provider` entry, there are no other reserved `Secret` entries.  In the interests of consistency, if a `Secret` includes any of the following entry names, the entry value **MUST** meet the specified requirements:
+
+| Name | Requirements
+| ---- | ------------
+| `host` | A DNS-resolvable host name or IP address
+| `port` | A valid port number
+| `uri` | A valid URI as defined by [RFC3986](https://tools.ietf.org/html/rfc3986)
+| `username` | A string-based username credential
+| `password` | A string-based password credential
+| `certificates` | A collection of PEM-encoded X.509 certificates, representing a certificate chain used in mTLS client authentication
+| `privateKey` | A PEM-encoded private key used in mTLS client authentication
+
+`Secret` entries that do not meet these requirements **MUST** use different entry names.
+
+# Application Projection
+
+A Binding `Secret` **MUST** be volume mounted into a container at in `$SERVICE_BINDINGS_ROOT/<binding-name>` with directory names matching the name of the binding.  Binding names **MUST** match `[a-z0-9\-\.]{1,253}`.  The `$SERVICE_BINDINGS_ROOT` environment variable **MUST** be declared and can point to any valid file system location.
+
+The `Secret` **MUST** contain a `kind` entry with a value that identifies the abstract classification of the binding.  It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` **MAY** contain any other entry.
+
+The contents of a secret entry may be anything representable as bytes on the file system including, but not limited to, a literal string value (e.g. `db-password`), a language-specific binary (e.g. a Java `KeyStore` with a private key and X.509 certificate), or an indirect pointer to another system for value resolution (e.g. `vault://production-database/password`).
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **MUST NOT** change during the lifetime of the container.
```

Comment:
This is not enforceable since [mounted Secrets are updated automatically](https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically)

Reply:
I think the spec should have a position where the Secret keys do not change, but the values can (i.e. an updated credential, etc).   This way the different implementation actors (bindable service, service binding processor, etc) all know that they should not change the Secret keys once mounting has occurred.

Reply:
@scothis's suggestion that softening this to **SHOULD NOT** seems right.  Make it clear what the correct behavior is from the spec, but acknowledge that in the real world this is unenforceable, and so an implementation shouldn't jump through hoops to stop it.  You just need to be defensive in case it does happen (unlikely in well-behaved systems).

Reply:
@scothis aye, atomicity of the update to the fs, doesn't guarantee the same of the read across multiple files. I suspect this is out of scope for the service binding operator, but it's certainly something to consider from an app layer. Detecting invalid state from an app perspective likely isn't something too trivial either.. (bad config may still work, but have unexpected side effects).